### DISCUSSION TITLE
Authorization Header 추가에 대해

### DISCUSSION THREAD

**Discussion by Choi-JJunho** [woowacourse-teams/2023-pium#145] [질문있어요] [answered]
> 해당 이슈를 작업하는 과정에서 발생한 의견입니다.
- #98 

## 의견

현재 저희 서비스 특성 상 대부분의 기능이 로그인을 해야 이용가능한 서비스로 구성되어있습니다.
로그인 기능까지는 구현하지 않더라도 Authorization 헤더에 인증정보를 넘기는 구조정도는 미리 가져가도 좋지 않을까요?

![image](https://github.com/woowacourse-teams/2023-pium/assets/49794401/91478f27-7d94-46ae-830f-8fdd0517f868)

## 추가 작업 예상내용

위 구조가 추가됨에 따라 각 트랙별로 발생할 것으로 예상되는 작업내용은 다음과 같습니다.

### 백엔드

Member 정보를 요구하는 API 작성 시 Parameter에 `@Auth` 어노테이션과 함께 Member를 추가한다.

```java
// 예시
@PostMapping
public ResponseEntity<Void> create(
        @RequestBody @Valid PetPlantRequest request,
        @Auth Member member) {
    PetPlantResponse petPlantResponse = petPlantService.create(request, member);
    return ResponseEntity.created(URI.create("/pet-plants/" + petPlantResponse.getId())).build();
}
```

### 프론트

Member 정보를 요구하는 API에 대해 요청을 보낼 때 Authorization 헤더에 사용자 인증정보(토큰)를 포함하여 보낸다.

```javascript
// 예시

fetch("http://api.pium.life/pet-plants",{
        method : "POST",
        headers : {
             Authorization : "pium@gmail.com"
        }
 })
.then(response => response.json())
.then(response => {
   console.log(response.data);
})
```

## 의견 재정리

위 추가작업에 대한 내용을 현재 개발과정에서 적용하는 것이 어떤지 여러분의 의견이 궁금합니다.

위 작업으로 인해 예상되는 이점은 다음과 같습니다.

- 향후 Google 로그인 과정을 도입하면서 Authorization 값만 수정하면 되기 때문에 향후 로그인 도입 과정에서 수정 소요가 많이 않을것이다.
- 유저 정보가 필요한 API를 미리 알고 향후 인증과 관련하여 일어날 수 있는 사이드이펙트를 예방할 수 있다.
- 보다 꼼꼼한 인수테스트를 수행할 수 있다.

**Comment by WaiNaat** [comment]
좋아요 👍 
토큰 내용은 아무거나 보내도 되는건가요?

**Reply by Choi-JJunho** [reply]
만약 적용하게 된다면 토큰 내용은 현재 DB에 담겨져있는 임의의 유저인 `pium@gmail.com` 이 되어야합니다!
```javascript
headers : {
  Authorization : "pium@gmail.com"
}
```

**Comment by yeonkkk** [comment]
좋은 생각인 것 같아요!
미리 해두면  이후 로그인 구현할 때 좀 더 수월할 것 같아요~

**Comment by Kim0914** [comment]
좋습니다 !

**Comment by sh111-coder** [comment]
안좋습니다! 폐기해주세요

**Reply by Kim0914** [reply]
감사합니다 !

**Comment by sh111-coder** [comment]
코드를 이렇게밖에 못짭니까???

**Reply by sh111-coder** [reply]
- by 필립

**Comment by rawfishthelgh** [comment]
저도 동의합니다! 나중에 변경할 부분이 줄어들 것 같아요

**Comment by Choi-JJunho** [comment, marked as answer]
해당 내용 관련하여 백엔드 코드 적용 완료했습니다.
- https://github.com/woowacourse-teams/2023-pium/pull/146
